pythonwhat/local.py

```diff
@@ -254,10 +260,20 @@ def run(state, relative_working_dir="", solution_dir="solution"):
                 has_printout(0)
             )
     """
-    # todo: configure these arguments automatically based on check_file info?
-    # once that is implemented, look into executing the file itself
+    # todo:
+    # look into executing the file itself
     # and keeping the process alive to extract values
-    sol_wd = Path(os.getcwd(), solution_dir, relative_working_dir)
+    if relative_working_dir is None:
+        if getattr(state, "path", False):
+            relative_working_dir = state.path.parent
+        else:
+            relative_working_dir = ""
```

Comment:
It works like this, but I do think it would be easier to reason about the code when you make this `Path(".")`.